dotnet/src/webdriver/Internal/Logging/FileLogHandler.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.IO;
+
+namespace OpenQA.Selenium.Internal.Logging
+{
+    /// <summary>
+    /// Represents a log handler that writes log events to a file.
+    /// </summary>
+    public class FileLogHandler : ILogHandler, IDisposable
+    {
+        // performance trick to avoid expensive Enum.ToString() with fixed length
+        private static readonly string[] _levels = { "TRACE", "DEBUG", " INFO", " WARN", "ERROR" };
+
+        private FileStream _fileStream;
+        private StreamWriter _streamWriter;
+
+        private readonly object _lockObj = new object();
+        private bool _isDisposed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileLogHandler"/> class with the specified file path.
+        /// </summary>
+        /// <param name="path">The path of the log file.</param>
+        public FileLogHandler(string path)
+        {
+            if (string.IsNullOrEmpty(path)) throw new ArgumentException("File log path cannot be null or empty.", nameof(path));
+
+            var directory = Path.GetDirectoryName(path);
+            if (!string.IsNullOrWhiteSpace(directory) && !Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+
+            _fileStream = File.Open(path, FileMode.OpenOrCreate, FileAccess.Write, FileShare.Read);
+            _fileStream.Seek(0, SeekOrigin.End);
+            _streamWriter = new StreamWriter(_fileStream, System.Text.Encoding.UTF8)
+            {
+                AutoFlush = true
+            };
+        }
+
+        /// <summary>
+        /// Handles a log event by writing it to the log file.
+        /// </summary>
+        /// <param name="logEvent">The log event to handle.</param>
+        public void Handle(LogEvent logEvent)
+        {
+            lock (_lockObj)
+            {
+                _streamWriter.WriteLine($"{logEvent.Timestamp:r} {_levels[(int)logEvent.Level]} {logEvent.IssuedBy.Name}: {logEvent.Message}");
```

Comment:
Instead of `r` timestamp format maybe better use `yyyy-MM-dd HH:mm:ss.fff`?

Reply:
Done.